### DISCUSSION TITLE
Link Prediction on Custom dataset

### DISCUSSION THREAD

**Discussion by bis1999** [pyg-team/pytorch_geometric#8114] [Q&A]
So I am working on doing some research on link prediction but on the basis of the graphical structure only i.e. the only information I have is that nodes and edges of most of the GNN Link prediction examples have node features(ex data.x), what should I replace it with? I was thinking of replacing it with an identity matrix with dimensions being a number of nodes. 

Also if you guys have any resources with link prediction that people are using their custom dataset instead of preloaded like CORA that would be really helpful

Thanks in Advance !

**Comment by wsad1** [comment]
It looks like the best thing to do would be to create node features, like node degreee, node centrality etc.
For more link prediction example check out the OGB link prediction tasks https://ogb.stanford.edu/docs/leader_linkprop/.

**Comment by rusty1s** [comment]
To follow-up on @wsad1, you can also learn an embedding matrix. This is identical to passing in the identity matrix but is actually more memory friendly, e.g.:
```python
embedding = torch.nn.Embedding(num_nodes, num_features)

def forward(self, n_id, edge_index):
    x = embedding(n_id)
    x = conv(x, edge_index)

**Reply by bis1999** [reply]
I see Thank you guys
